app/services/open_library_service.rb

```diff
@@ -0,0 +1,28 @@
+##
+# To see format, jscmd and fields options go to page
+# https://openlibrary.org/dev/docs/api/books
+##
+
+class OpenLibraryService
+  include HTTParty
+  base_uri 'openlibrary.org/api/books'
+
+  def initialize(format = 'json', jscmd = 'data', fields = %w[title subtitle authors pagination])
+    @format = format
+    @jscmd = jscmd
+    @fields = fields
+  end
+
+  def book_info(isbn)
```

Comment:
You may raise an Exception instead of returning the error, if you don't do that there is no way you can know if the response is an error response or not.